core/src/main/java/tc/oc/pgm/controlpoint/ControlPoint.java

```diff
@@ -311,6 +311,7 @@ protected void tickCapture(Duration duration) {
       }
     }
 
+    boolean contested = false;
     int lead = 0;
```

Comment:
I fail to see how `lead` and `contested` represent any different state in this whole block of code.
A lead of > 0, means there's a leader team, so, the point is *not* contested, and that already takes into account the rules required to make it work.
There is one exception to this, and that is when there's no one on the point at all, so lead is naturally 0 in `playerCounts.get(leader)`.

But that essentially means, contested is always equal to `playerCounts.get(leader) > 0  && lead <= 0` (evaluated *after* the switch statement), or alternatively simply, `leader != null  && lead <= 0` because leader is only set if *some player* is dominating.